lms/resources/_js_config/__init__.py

```diff
@@ -68,14 +68,22 @@ def add_document_url(self, document_url):
                 ),
             }
         elif document_url.startswith("vitalsource://"):
-            svc = self._request.find_service(VitalSourceService)
+            svc: VitalSourceService = self._request.find_service(VitalSourceService)
+
+            if svc.sso_enabled:
+                # nb. VitalSource doesn't use Via, but is otherwise handled
+                # exactly the same way by the frontend.
+                self._config["viaUrl"] = svc.get_sso_redirect(
+                    user_reference=self._request.lti_params[svc.user_lti_param],
+                    document_url=document_url,
+                )
+            else:
+                # This looks a bit silly, but pretty soon the above will
+                # be setting `api.viaURL` not `viaURL`
+                self._config["viaUrl"] = svc.get_book_reader_url(
+                    document_url=document_url
+                )
```

Comment:
In the next PR (once rebased) this will be using different config locations as one will be setting a callback